Classic tiegcm dev

timhoar opened the pull request.

timhoar committed:
First stab at running one-cycle-at-a-time

Untested at this point.

timhoar committed:
revert to Intel compilation

timhoar committed:
conveying ensemble size to job array

gwbowden commented:
Sorry to take so long to act on this. I would like to keep the async4 scripts in addition to the ones using the new approach,

For now I've copied the new scripts into a new directory on Gadi. There have been a number of issues, but as things stand I've created a run directory for the experiment and am trying to run `assimilate.csh` and `advance_tiegcm.csh`.

Currently `assimilate.csh` fails with the message:
`ERROR: inflation staging failed.`
indicating that neither `priorinf_pointer_file.txt` nor `prior_inflate_ics` are available. How can I find/generate the required file (or should I bypass this part of the script)?

`advance_tiegcm.csh` fails with:
`ls: cannot access 'dart_restart': No such file or directory`
How should I generate this file for testing? Will it be produced by first running `assimilate.csh`?

timhoar commented on models/tiegcm/shell_scripts_gadi/assimilate.csh:
```diff
@@ -0,0 +1,211 @@
+
+# Must copy each filter_restart.nnnn to the right place, update link
+# so that 'dart_to_model' gets what it needs.
+foreach RESTART ( filter_restart.???? )
+   set instance = $RESTART:e
+   set INSTANCE_DIRECTORY = `printf "instance_%04d" $instance`
+   cd $INSTANCE_DIRECTORY
+   ${MOVE} -v ../$RESTART  $RESTART.$TIMESTAMP
+   \unlink dart_restart
+   ${LINK} $RESTART.$TIMESTAMP dart_restart
+   cd ..
+end
```
After this loop finishes, there should be a `dart_restart` in each of the instance directories.  Is this not happenning?

timhoar reviewed (commented):
George ... what other modifications have you made to the scripts? I need to see them to be able to put things in context.

timhoar committed:
adding support for advance_time in tiegcm

timhoar committed:
staging works with cheyenne environment

timhoar committed:
prototype assimilate script

Creates an inflation file if one does not exist.
Does not use obs_seqence tool to cut out desired observations
Also needs to prepend an 'advance_to_time' before dart_to_model

timhoar committed:
subsets desired obs on-the-fly

timhoar committed:
removed deprecated svn keywords

timhoar committed:
staging and assimilation complete

The assumption here is the there is a single large observation
sequence file that gets processed into a 1-hour assimilation-sized
file. The 'advance_to_time' record is prepended to the output of
filter so 'dart_to_model' (in advance_tiegcm.csh) correctly creates
the 'namelist_update' file.

timhoar committed:
The processing cycle should be working.

Next step is to replace the job array with whatever is recommended
by the gadi administrators.

timhoar commented:
George ... this should work with very little tweaking ... save that I have not replaced the job array script yet.
I changed a lot of the Fortran to clean up remnants from the svn-to-git conversion (the svn keywords are not supported by git)
so those are just nuisance changes. If you have not worked through a pull request, let me know and we can go through it together. 

The storm is living up to they hype - even though it got a slow start. Point being ... I'm here ...
